cache.go

```diff
@@ -87,15 +90,37 @@ func (cd *cachedData) flush() {
 		return
 	}
 
-	file, err := os.OpenFile(cd.Path, os.O_RDWR|os.O_TRUNC, 0666)
+	file, err := fileutil.LockFile(cd.Path, os.O_RDWR, 0666)
 	if err != nil {
 		log.Printf("Cannot write to the cache file: %v", err)
```

Comment:
I'd also block, just to mimic what zypper does :) Moreover:

- As @monstermunchkin says, we wouldn't block the user for too much.
- Having multiple instances of zypper-docker running is already a corner case which won't happen often I think.